crates/loader/src/local/assets.rs

```diff
@@ -40,36 +44,54 @@ pub(crate) async fn prepare_component(
 pub struct FileMount {
     /// The source
     pub src: PathBuf,
+    /// The relative path to spin.toml parent folder.
+    pub relative_src: String,
```

Comment:
Do you need to keep this in the FileMount struct?  The excluded files filter should be able to work on absolute paths - just prepend the base source directory (which we call `rel`, but is absolute) before resolving each pattern (as we do in `collect_pattern`).

Reply:
Thanks for the suggestion. I remove `relative_src` and updates in `collect_placement`. It looks much better now!